src/uu/df/src/filesystem.rs

```diff
@@ -67,6 +67,16 @@ where
     } else {
         path.as_ref().to_path_buf()
     };
+
```

Comment:
Why not do something like this instead ? 

```rust
    let matches: Option<&MountInfo> = mounts
        .iter()
        .find(|mi| mi.dev_name.eq(&path.to_string_lossy()));

    matches.or_else(|| {
        mounts
            .iter()
            .filter(|mi| path.starts_with(&mi.mount_dir))
            .max_by_key(|mi| mi.mount_dir.len())
    })
}
```

Reply:
maybe we should use the stater module for it too while we are at it !

Reply:
Thanks for your suggestion. This is indeed a much cleaner solution and I switched to it in the recent push.